components/raftstore/src/store/peer.rs

```diff
@@ -1498,6 +1503,10 @@ where
             "term" => term,
             "peer_id" => self.peer_id(),
         );
+
+        self.read_progress
+            .update_leader_info(leader_id, term, self.region());
```

Comment:
`region` may not be correct here. If it's an uninitialized peer, region's peer list can be empty.

Reply:
Yes, but we only use the leader's peer list. The `leader_info` will also be updated after snapshot applied and epoch changed, so the peer's `leader_info` will be corrected as it catches up with more data.

Reply:
I don't think so. `update_leader_info` is only called when leadership changes, so even for leader, the peer list can become wrong after applying configuration change.

Reply:
Actaully,  `update_leader_info` is called inside `Peer::set_config`, `Peer::on_leader_changed` and `PeerFsmDelegate::on_ready_apply_snapshot`, it has cover all cases where the peer list, region epoch and leader id will change.